scripts/sql/hssc_obs_fact_etl.sql

```diff
@@ -0,0 +1,404 @@
+        '@', '@', null, null, '@', '@', '@', sysdate, 
+        px.sourcesystem_cd, null, '1' , null, null);
+                    
+          EXCEPTION WHEN OTHERS 
+          THEN
+          pkg_error.log(p_error_code => substr(sqlerrm,1,9),
+              p_error_message => substr(sqlerrm,12) || '.EXCP ' 
+              || px.encounter_num || ', ' || px.patient_num,
+              p_package => '', p_procedure => m_procname);
+    END;       
+        m_rowcnt := m_rowcnt + 1;
+
+        if ( m_rowcnt > 0 and mod(m_rowcnt, m_comrows) = 0 ) then
+              COMMIT;
+        end if;
+  end loop;
+  
+  commit;
+
+END ETL_PROCEDURE;
+
+-------------------------------------------------------------------------------------------------
+
+create or replace PROCEDURE          ETL_LABS authid current_user
+IS
+
+   m_rowcnt NUMBER := 0; -- row counter
+   m_comrows NUMBER := 40000; -- commit every m_comrows
+   m_procname ERRORLOG.PROCEDURE_NAME%TYPE := 'ETL_LABS';
+
+
+  cursor labs_cur is
+    SELECT DISTINCT 
+      /*+ PARALLEL 4 */
+        S.VISIT_DEID as ENCOUNTER_NUM,
+        S.PATIENT_DEID as PATIENT_NUM,
+      'LOINC:' || X.LOINC_CD as CONCEPT_CD, --how do we pull RZ.TGT_CODE_TYPE?
+    
+      COALESCE(X.RESULT_DT, V.VISIT_START_DATE) - S.SHIFTVALUE as START_DATE, -- Lab Date or Visit Start Date -- modified
+      '1' as INSTANCE_NUM, 
+    
+      COALESCE(X.result_DT, V.VISIT_END_DATE) - S.SHIFTVALUE as END_DATE, -- Lab Date or Visit End Date,  -- modified
+    
+      X.DATASOURCE_ID as SOURCESYSTEM_CD
+      FROM cdw.VISIT_DEID_MAP_HSSC@dtdev S
+      JOIN cdw.VISIT@dtdev V ON (V.VISIT_ID = S.VISIT_ID)
+      JOIN cdw.lab_result@dtdev X ON (V.HTB_ENC_ACT_ID = X.HTB_ENC_ACT_ID)
+
+      -- remove ones without LOINC codes
+      WHERE X.LOINC_CD is not null; -- added 4/19/2016
+
+
+  BEGIN
+
+
+    for lb in  labs_cur LOOP
+          BEGIN
+           insert into i2b2hsscdata.observation_fact 
+           (encounter_num,patient_num,concept_cd,provider_id,start_date,
+             end_date, modifier_cd,instance_num,valtype_cd,tval_char,
+             nval_num,quantity_num,valueflag_cd,units_cd, location_cd,
+             import_date,sourcesystem_cd,observation_blob,
+             confidence_num,download_date,update_date)
+          values (lb.encounter_num,lb.patient_num,lb.concept_cd,
+          '@',lb.start_date,lb.end_date,'@',lb.instance_num, '@', '@', 
+          null, null, '@', '@', '@',sysdate,lb.sourcesystem_cd, 
+          null, '1', null, null);
+        
+          EXCEPTION WHEN OTHERS 
+          THEN
+          pkg_error.log(p_error_code => substr(sqlerrm,1,9),
+              p_error_message => substr(sqlerrm,12) || '.EXCP ' 
+              || lb.encounter_num || ', ' || lb.patient_num,
+              p_package => '', p_procedure => m_procname);
+      
+        m_rowcnt := m_rowcnt + 1;
+
+        if ( m_rowcnt > 0 and mod(m_rowcnt, m_comrows) = 0 ) then
+              COMMIT;
+        end if;
+        END;
```

Comment:
good catch. we have other changes we did yesterday - collecting all of those (including patricia's changes to cursor select etc.